src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-form/rgw-bucket-form.component.ts

```diff
@@ -282,7 +282,7 @@ export class RgwBucketFormComponent extends CdForm implements OnInit {
             return false;
           }
           // Bucket names can contain lowercase letters, numbers, and hyphens.
-          if (!/[0-9a-z-]/.test(label)) {
+          if (!/^\S*$/.test(name) || !/[0-9a-z-]/.test(label)) {
```

Comment:
With regex, if you have to add multiple checks it usually means you're not doing it right. In this very case, you need to add more constrains: when the match starts (^), when it ends ($) and the character repetition (+, one or more):
```suggestion
          if (!/^[0-9a-z-]+$/.test(label)) {
```

Reply:
I put two conditions because one is checking the whole name while other is checking the last character which is entered. Does that makes sense? @epuertat

Reply:
You can even check the allowed length: `if (!/^[0-9a-z-]{3,63}$/.test(label)) {`

Reply:
I can do this. But we already checking this just little bit upward. And if we do it like this, then we will be loosing error message according to the corresponding error. That's why I separated it. Infact it was one of the RFEs that I previously addressed.

Reply:
Ok got it. I feel that this whole validation is a bit overdesigned. TBH the [S3 validation rules ](https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html) are not an example of clear concise rules (an IPv4 address matches perfectly the other rules... but why are we checking for IPv6, when ":" is not supported?).

Reply:
BTW [here you can find a single regex](https://stackoverflow.com/questions/50480924/regex-for-s3-bucket-name) to match everything together... We'd loss the ability to report specific failures, but we could simply print the aggregated validation rules... In any case, given we have all this code already there, let's keep the approach.

Reply:
I've seen that regex when I was doing the initial async validator cleanup. But since we needed to report the specific failures I didn't implemented it. (Although it would have reduced the code.)